### ISSUE TITLE
3.x to 4.0 converter does not convert rotation_degrees property of animations

### ISSUE THREAD

**Issue by sketchyfun** [godotengine/godot#65481]
### Godot version

4.0 Alpha 16

### System information

Windows 10

### Issue description

The rotation_degrees property has been removed from spatial nodes, however the 3.x -> 4.0 converter does not update animations from 3.x projects that used the rotation_degrees property and convert them into rotation (radians)

### Steps to reproduce

Open the attached project in 4.0 and once it's been converted, check the animation player in TestScene.tscn You'll see that the rotation_degrees property is still present on the timeline, and is an invalid property

### Minimal reproduction project

[animation_rotation_degrees.zip](https://github.com/godotengine/godot/files/9507441/animation_rotation_degrees.zip)

**Comment by Zireael07** [comment]
IIRC already reported, can confirm.

**Comment by sketchyfun** [comment]
> IIRC already reported, can confirm.

I searched a couple of times for stuff relating to rotation_degrees, but I wasn't able to find anything about the animation side of things

**Comment by Calinou** [comment]
Related to https://github.com/godotengine/godot/issues/63390 (which proposes a different solution).

**Comment by Lodugh** [comment]
Its not just animations of spatial nodes but also 2D nodes, that have this issue.

**Comment by Calinou** [comment]
> Not sure why rotation_degrees was even removed?

This is described in detail in https://github.com/godotengine/godot/issues/63390. There were two properties effectively doing the same thing.

> The 3 > 4 converter also does not convert rotation_degrees in code to "rotation" and does not convert "deg2rad" to the new "deg_to_rad"

The converter only renames property renames, as this is the 90% use case. It can't transform existing properties that need some kind of conversion with a multiplier. Fixing this in scripts in particular would be quite difficult to do correctly.

**Comment by Zireael07** [comment]
The converter also doesn't deal with animations, as already mentioned

**Comment by fracteed** [comment]
I did read that thread and totally agree with the OP and his thoughts on this. I have no issue with removing "_degrees" from the name, but it should have remained as functioning in degrees. It is going to cause a lot of confusion by having the ui parameter in degrees, but having to use radians in code.

We already had the set_rotation method to use radians so now there are 2 methods for setting rotation in radians, but none for explicitly using degrees. Of course it can be worked around, but code becomes messier with all the conversions. Either way, the converter should change animations to the new system when porting from 3.x

**Comment by sketchyfun** [comment]
It's also a bit annoying to be told to just use radians with TAU or a converter function when the editor side of things is handled in degrees, which shows that it's what most people expect to work in. I get that a lot of programmers might be used to working in radians in code, but many game developers aren't professional coders and would be far more used to describing things in degrees. Having both properties available just made a lot more sense to me as then everyone has a choice, and it doesn't appear to have removed any bloat/complexity from the engine anyway. For most people degrees are just easier/more natural to work with as that's what they've been taught to use growing up.

**Comment by tinmanjuggernaut** [comment]
> The converter only renames property renames, as this is the 90% use case. It can't transform existing properties that need some kind of conversion with a multiplier. Fixing this in scripts in particular would be quite difficult to do correctly.

I have `rotation_degrees` animations that are broken. I have one where I rotate a wheel from 0 to -359.9 degrees. When I remade the track using `rotation`, the values were identical. The GD3 and GD4 editors only work in degrees. The only change was moving the track from `rotation_degrees` to `rotation`. It could have just renamed the track.

However, I'd say this is a duplicate of #63390 or redundant. Fixing the converter isn't the real problem. That's just a band-aid. I rewrote my issue to clarify the core points and made sure this issue was mentioned and referenced.

The core issues are the inspector and code work in different units and we cannot use degrees in code. The Inspector is degrees only, Code is radians only. Both units are currently implemented in the engine, so there is absolutely no code savings by neutering the API. This was just a very poor design choice that forces us to move conversions from C++ to GDscript and redo a lot of work for no benefit.

**Comment by sketchyfun** [comment]
(sorry, commented on mobile and accidentally closed the issue!)

**Comment by reduz** [comment]
I re-added this property in 4.0 for code compatibility so the animations will work again, however you will not be able to insert new keyframes as the property is no longer exposed in the editor.

So, this is issue is fixed to some extent, but conversion should ideally be better.

**Comment by sketchyfun** [comment]
I've tried the latest RC builds, and although rotation_degrees is now available, existing animations still can't use it since it's not exposed in the inspector. I thought existing tracks that use it would work, but it seems like it fails and ends up turning all the original keyframes into 'dummy' keyframes (I guess the base keyframe class since it can't find the appropriate type, since it thinks the property doesn't exist)

**Comment by akien-mga** [comment]
To be clear we understand things the same way. As of 4.0 RC 5 (likely anything after #70263), _existing animations_ using `rotation_degrees` will work fine in Godot 4.0, as in you can still play the animation and it works:

![image](https://user-images.githubusercontent.com/4701338/221434402-b64fa2be-fdb5-4802-bf84-f29dcc20becb.png)

What's missing is that since the `rotation_degrees` property is not exposed in the inspector, the pre-existing values can be modified:
![image](https://user-images.githubusercontent.com/4701338/221434479-76f9df14-3494-4ba8-828c-7ad7f3ab9500.png)

And while keyframes can be added with a right click, they also can't be given a value.

So there's no loss of data, but loss of the ability to edit those tracks further. Trying to keyframe the `rotation` property adds a new track.

So to solve this, either the `rotation_degrees` animations should be converted to `rotation` so they stay editable, or the `rotation_degrees` property should be shown (conditionally?) when editing an animation which contains such track (probably not trivial, and maybe not the best UX either).

**Comment by sketchyfun** [comment]
I haven't tried RC 5, but when I wrote the last comment the tracks using rotation_degrees didn't animate at all, I'll have to try a newer version.

As for fixing the issue, solving it with the convertor seems like the most sensible idea, or possibly a temporary AnimationTrackKeyLegacyRotationEdit (catchy!) to handle these until a later date.

In the mean time I've compiled godot myself with the rotation_degrees property re-exposed which thankfully was only a one line change
